examples/generated-src/objc/TXSItemList.h

```diff
@@ -4,7 +4,9 @@
 #import <Foundation/Foundation.h>
 
 @interface TXSItemList : NSObject
-- (nonnull instancetype)initWithItems:(nonnull NSArray<NSString *> *)items;
+- (nonnull instancetype)init NS_UNAVAILABLE;
++ (nonnull instancetype)new NS_UNAVAILABLE;
+- (nonnull instancetype)initWithItems:(nonnull NSArray<NSString *> *)items NS_DESIGNATED_INITIALIZER;
```

Comment:
@finalpatch The generator does not 'generate' `init` or `new`, but they are inherited from `NSObject`. If you have a record with any non-optional fields, and initialize it say `[TSXItemList new]`, it will throw an implicitly unwrapped optional runtime exception in Swift on accessing `items`. 

Djinni `records` are read-only as well, so it's not possible to set non-optional properties after construction anyway. 

I've also added `NS_DESIGNATED_INITIALIZER` which makes the compiler enforce that any convenience constructor must call this method.

Reply:
Thanks for the explanation, this make sense.  I didn't know we have always been able to default init records with NSObjct's constructor.

Reply:
No prob! We found this the hard way (in a test case thankfully) but I figured it was a quick and easy patch and modernizes the generated Obj-C code (especially for Swift consumption).